app/javascript/packs/shared/data_table/DataTable.jsx

```diff
@@ -0,0 +1,69 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { BootstrapTable, TableHeaderColumn } from 'react-bootstrap-table';
+
+class DataTable extends Component {
+  static cleanColumnAttributes(col) {
+    // Remove additional attributes from the columns
+    const {
+      id, isKey, textId, name, position, visible,
+      sortable, locked, ...cleanCol
+    } = col;
+    return cleanCol;
+  }
+
+  constructor(props) {
+    super(props);
+    this.cleanProps = this.cleanProps.bind(this);
+    this.displayHeader = this.displayHeader.bind(this);
+  }
+
+  cleanProps() {
+    // Remove additional props from the props value
+    const {columns, ...cleanProps} = this.props;
+    return cleanProps;
+  }
+
+
+  displayHeader() {
+    const orderedCols = [...this.props.columns].sort((a, b) => a.position - b.position);
```

Comment:
You don't need the spread operator here https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Operators/Spread_operator

here is babel repl you can use it to see how the es6 code is converted to es5 https://babeljs.io/repl/